server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstraction.java

```diff
@@ -210,6 +217,13 @@ public boolean isHidden() {
             return isHidden;
         }
 
+        @Override
+        public boolean isSystem() {
+            //G-> This is probably not the best way to tell if an alias is a system-index-alias
+            // this should probably be checked on the alias layer and added as a property in the AliasMetadata
+            return referenceIndexMetadatas.stream().anyMatch(IndexMetadata::isSystem);
```

Comment:
I agree and think we should make this similar to what we did with hidden aliases